cloud/cloud.js

```diff
@@ -857,50 +857,205 @@ Moralis.Cloud.define(
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const responses = getResponses(taskId);
+    return responses;
+  },
+  {
+    fields: ["taskId"],
+    requireUser: true,
+  }
+);
 
-      res.set("taskerRating", rating);
-      res.set("hasRated", true);
+/* ------------------------------------------------------------------- */
 
-      await res.save();
+Moralis.Cloud.define(
+  "getTaskUsers",
+  async (request) => {
+    async function getTaskUsers(taskId) {
+      tableName = "TaskUsers";
+      const TaskUsers = Moralis.Object.extend(tableName);
+      const query = new Moralis.Query(TaskUsers);
+      query.equalTo("taskId", taskId);
+      query.notEqualTo("status", 0);
+      const res = await query.find();
+      return res;
     }
-
+    const taskId = request.params.taskId;
     const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const taskUsers = getTaskUsers(taskId);
+    return taskUsers;
+  },
+  {
+    fields: ["taskId"],
+    requireUser: true,
+  }
+);
+
+Moralis.Cloud.define(
+  "getUserResponse",
+  async (request) => {
+    async function getTaskUsers(taskId, taskerId) {
+      tableName = "Responses";
+      const Responses = Moralis.Object.extend(tableName);
+      const query = new Moralis.Query(Responses);
+      query.equalTo("taskId", taskId);
+      query.equalTo("userId", taskerId);
+      const res = await query.find();
+      return res;
+    }
     const taskId = request.params.taskId;
-    const rating = Number(request.params.rating);
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId))) return [];
+    const userResponse = getTaskUsers(taskId, request.params.taskerId);
+    return userResponse;
+  },
+  {
+    fields: ["taskId", "taskerId"],
+    requireUser: true,
+  }
+);
 
-    if (!(await checkTaskerTaskHasNotRated(ethAddress, taskId)))
+Moralis.Cloud.define(
+  "updateApprovalStatus",
+  async (request) => {
+    const taskId = request.params.taskId;
+    const ethAddress = request.user.get("ethAddress");
+    if (!(await checkRequesterCreatedTask(ethAddress, taskId)))  {
       return {
         success: false,
-        message: `Address ${ethAddress} cannot rate task ${taskId}.`,
+        message: "Approval update failed - unauthorized update!"
       };
+    }
+    tableName = "TaskUsers";
+    const TaskUsers = Moralis.Object.extend(tableName);
+    const query = new Moralis.Query(TaskUsers);
+    return query.get(request.params.objectId).then(
+      (taskUser) => {
+        if (taskUser.get("status") !== 1) {
+          return {
+            success: false, 
+            message: "Cannot update status of tasks that are already approved/rejected"
+          };
+        }
+        taskUser.set("status", request.params.newStatus)
```

Comment:
Can you first check that `request.params.newStatus` is either `2` or `3`, and if not then return an error, e.g.

> Task status must be updated to either "approved" or "rejected"